### PR TITLE
Binary serialization unused methods and attributes cleanup and misc code cleanup

### PR DESCRIPTION
**Pull request by ViktorHofer**
Not sure about:
- https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Diagnostics/Stacktrace.cs If I remove the serialization methods, `rgMethodHandle` will never be set but its value is used in `GetMethodBase(int)`. So is the whole StackFrameHelper type maybe tied to serialization?

Also did some other misc code cleanup / comment cleanup for serialization

### CODE REVIEW

File: src/mscorlib/src/System/IO/StreamReader.cs [hunk 1]
```diff
@@ -216,14 +213,14 @@ public StreamReader(String path, Encoding encoding, bool detectEncodingFromByteO
 
             _checkPreamble = (_preamble.Length > 0);
             _isBlocked = false;
-            _closable = !leaveOpen;
+            _leaveOpen = !leaveOpen;
```
Comment (stephentoub) [inline src/mscorlib/src/System/IO/StreamReader.cs]: "closable" has the opposite meaning of "leaveOpen".  If you're going to change the name, you need to change conditions like this, too.  The leaveOpen functionality in general now appears to be broken.

File: src/mscorlib/src/System/Resources/ResourceManager.cs [hunk 2]
```diff
@@ -99,9 +98,8 @@ internal class PRIExceptionInfo
     // RM will need to use file-based or manifest-based resources, and 
     // afterwards refers to this through the interface IResourceGroveler.
     // 
-    // Serialization Compat: Ideally, we could have refactored further but
-    // this would have broken serialization compat. For example, the
-    // ResourceManager member UseManifest and UseSatelliteAssem are no 
+    // TODO: more serialization cleanup as type isn't serializable anymore.
```
Comment (stephentoub) [inline src/mscorlib/src/System/Resources/ResourceManager.cs]: You've added a TODO... is there an issue number associated with this?

Comment (ViktorHofer) [inline src/mscorlib/src/System/Resources/ResourceManager.cs]: Good point, not yet. Let me first discuss that matter with the area owner.

@tarekgh the issue guide says you are area owner of System.Resources. The comments in ResourceManager suggest that code refactoring could/should be done when serialization support is stripped from the type. This is now the case. Is the comment still accurate, are there optimizations possible now?

Comment (tarekgh) [inline src/mscorlib/src/System/Resources/ResourceManager.cs]: I am not aware of any clean up needed in ReourceManager nor aware of any issue around the serialization. it looks the comment is really old comment.

---
In reply to: [131532032](https://github.com/dotnet/coreclr/pull/13242#discussion_r131532032) [](ancestors = 131532032)

**Comment by ViktorHofer** [comment]
@dotnet-bot test this please

**Comment by ViktorHofer** [comment]
I'm testing something here, don't wonder that my commits/work is gone.

**Comment by karelz** [comment]
@ViktorHofer are you still working on this one? 2 weeks no update ;)

**Comment by ViktorHofer** [comment]
Yes, I'm still working on it. And I can't close this as I need CI for it. I will continue to work on it now.

**Comment by ViktorHofer** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by ViktorHofer** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by jkotas** [comment]
The OSX failure is known issue: https://github.com/dotnet/coreclr/issues/13779

**Comment by ViktorHofer** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by ViktorHofer** [comment]
@jkotas can you please review my changes? Three more non serializables types contain NonSerializable attributes but I will tackle them in another PR as they are causing tests here in coreclr to fail.

File: src/mscorlib/src/System/IO/IOException.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+**
+**
+===========================================================*/
+
+using System;
+using System.Runtime.Serialization;
+
+namespace System.IO
+{
+    public class IOException : SystemException
```
Comment (jkotas) [inline src/mscorlib/src/System/IO/IOException.cs]: This file has moved to shared partition. You should do the fix there, and not adding a dead copy of the file here.

**Comment by ViktorHofer** [comment]
@jkotas done. Dan did already the necessary cleanup in IOException.

**Pull request merged**